File: backend/classe/voie.py
class Voie:
    nombre_d_elements = 0

    def __init__(self, Gare1, Gare2, ligne, temps, Bidirectionnel=1):
        self.Gare1 = Gare1
        self.Gare2 = Gare2
        self.ligne = ligne
        self.temps = temps
        self.Bidirectionnel = Bidirectionnel #Si egale a 0 alors Gare1 -> Gare 2 (besoin pour la ligne 10)
        Gare1.add_Voie(self)
        Gare2.add_Voie(self)
        Voie.nombre_d_elements += 1

    def get_attr(self, attr_name):
        return getattr(self, attr_name, None)

    def get_other(self, gare):
        if gare == self.Gare1:
            return self.Gare2
        elif gare == self.Gare2:
            return self.Gare1
        else:
            return None

    def __str__(self):
        return (f"Voie(Gare1={self.Gare1.get_attr('name')}, "
                f"Gare2={self.Gare2.get_attr('name')}, "
                f"ligne={self.ligne.get_attr('name')}, temps={self.temps}, "
                f"Bidirectionnel={self.Bidirectionnel})")

    #compare all element with a second voie
    def __eq__(self, other):
        print("a")
        return self.Gare1 == other.Gare1 and self.Gare2 == other.Gare2 and self.ligne == other.ligne and self.temps == other.temps and self.Bidirectionnel == other.Bidirectionnel

File: backend/src/Voie.java
public class Voie {
    private static int nombreDElements = 0;

    private Gare gare1;
    private Gare gare2;
    private Line ligne;
    private int temps;
    private int bidirectionnel;

    public Voie(Gare gare1, Gare gare2, Line ligne, int temps, int bidirectionnel) {
        this.gare1 = gare1;
        this.gare2 = gare2;
        this.ligne = ligne;
        this.temps = temps;
        this.bidirectionnel = bidirectionnel;
        gare1.addVoie(this);
        gare2.addVoie(this);
        nombreDElements++;
    }
    public Gare getOther(Gare gare) {
        if (gare == gare1) {
            return gare2;
        } else if (gare == gare2) {
            return gare1;
        } else {
            return null;
        }
    }

    @Override
    public String toString() {
        return String.format("Voie(Gare1=%s, Gare2=%s, ligne=%s, temps=%d, Bidirectionnel=%d)",
                gare1.getName(), gare2.getName(), ligne.getName(), temps, bidirectionnel);
    }

    @Override
    public boolean equals(Object obj) {
        if (this == obj) return true;
        if (obj == null || getClass() != obj.getClass()) return false;
        Voie other = (Voie) obj;
        if (gare1.equals(other.gare1) && gare2.equals(other.gare2) && ligne.equals(other.ligne) && temps == other.temps && bidirectionnel == other.bidirectionnel){
            return true;
        }
        if (gare1.equals(other.gare2) && gare2.equals(other.gare1) && ligne.equals(other.ligne) && temps == other.temps && bidirectionnel == other.bidirectionnel){
            return true;
        }

        return false;
    }

    public static int getNombreDElements() {
        return nombreDElements;
    }

    public Line getLigne() {
        return ligne;
    }

    public int getTemps() {
        return temps;
    }

}
